src/PlugWallet/ledger.ts

```diff
@@ -0,0 +1,177 @@
+import { polling, PublicKey } from "@dfinity/agent";
+import { BinaryBlob, blobFromBuffer, blobFromUint8Array } from "@dfinity/candid";
+import { Delegation, DelegationChain, DelegationIdentity } from "@dfinity/identity";
+import { Principal } from "@dfinity/principal";
+import { NFTDetails, TokenInterfaces } from "@psychedelic/dab-js";
+
+import { Address } from "../interfaces/contact_registry";
+import { JSONWallet, PlugWalletArgs } from "../interfaces/plug_wallet";
+import { createAgent } from "../utils/dfx";
+import { GenericSignIdentity } from "../utils/identity/genericSignIdentity";
+import Secp256k1KeyIdentity from "../utils/identity/secpk256k1/identity";
+import PlugWallet from "./base";
+import proto from "../utils/proto";
+import { LEDGER_CANISTER_ID } from "../utils/dfx/constants";
+import { validateAccountId } from "../PlugKeyRing/utils";
+import { getAccountId } from "../utils/account";
+
+const { AccountIdentifier, BlockHeight, ICPTs, Memo, Payment, SendRequest, Certification } = proto.ic_ledger.pb.v1;
+
+class PlugWalletLedger extends PlugWallet {
+    protected ledgerIdentity: GenericSignIdentity;
+
+    constructor(args: PlugWalletArgs) {
+      super(args);
+      this.ledgerIdentity = args.identity;
+      this.identity = Secp256k1KeyIdentity.ephimeral();
+      this.agent = createAgent({fetch: this.fetch, defaultIdentity: this.identity});
+    }
+
+    public async sign(payload: BinaryBlob): Promise<BinaryBlob> {
+      return this.ledgerIdentity.sign(payload);
+    }
+
+    public async transferNFT(args: {
+      token: NFTDetails;
+      to: string;
+    }): Promise<boolean> {
+      throw new Error('Transfer NFT not supported on Ledger');
+      return this.executeWithDelegation(super.transferNFT, args);
+    };
+
+    public async burnXTC(args: { to: string; amount: string }) {
+      throw new Error('Burn XTC not supported on Ledger');
+      return this.executeWithDelegation(super.burnXTC, args);
+    };  
+    
+    public async send(args: {
+      to: string;
+      amount: string;
+      canisterId: string;
+      opts?: TokenInterfaces.SendOpts;
+    }): Promise<TokenInterfaces.SendResponse> {
+      if (args.canisterId !== LEDGER_CANISTER_ID) {
+        throw new Error(`Token ${args.canisterId} not supported on Ledger`)
+      }
+      const to = validateAccountId(args.to) ? args.to : getAccountId(Principal.fromText(args.to));
+
+      if(!validateAccountId(to)) {
+        throw new Error('Invalid Account ID')
+      }
+
+      const canisterId = Principal.fromText(LEDGER_CANISTER_ID);
+
+      const agent = createAgent({defaultIdentity: this.ledgerIdentity});
+
+      const amount = args.amount; // 0.1 ICP
+      const memo = args.opts?.memo || BigInt(0);
+      const maxFee = args.opts?.fee || BigInt(10000);
```

Comment:
what if `fee` is equal to 0, you expect it to be true? in that case you should use `??` instead `| |` otherwise cool